### PR TITLE
Add ability to pull student repos issue #8

### PR DESCRIPTION
**Pull request by shanep**
### What are you trying to accomplish?

Change the default behavior when cloning student repos to skip any repos that have already been cloned. Add in a command line flag to allow existing repos to be sync'd instead of a full clone.

Fix issue #14
Implements Feature #8

### What approach did you choose and why?

Instead of doing a fresh clone every single time, skip any repos that have already been cloned and give the user a new command line flag to sync. If the user wants a fresh clone then they can already use the --directory flag to do so.

UPDATE: This change should take some pressure off your endpoint :)

### What should reviewers focus on?

I couldn't see anywhere in the documentation where the **gh** cli allows you to pass a path when performing a **sync** so I had to change the working directory. If there is a better way to do a sync with the gh client please let me know :)

### CODE REVIEW

**Review by NukeManDan — APPROVED** [review]
Will help a lot for using this tool for sure! :heart_eyes:

**Review by zrdaley — CHANGES_REQUESTED** [review]
@shanep thank you for this contribution!

I was chatting with @RyanHecht and in order to be consistent with typical git commands, it would make more sense to have a `gh classroom pull` command rather than a `--sync` flag on the `gh classroom clone` command. 

If you could take some time to make this into it's own command then I'd be happy to re-review.

**Comment by shanep** [comment]
@zrdaley no problem! I like that approach its keeps the semantics all lined up and doesn't overload the clone command :) I will get it all updated as soon as I can :)

**Review by shanep — COMMENTED** [review]
Changes have been pushed to the branch and are ready for review :)

File: cmd/gh-classroom/clone/student-repos/student-repos.go [hunk 1]
```diff
@@ -89,11 +90,15 @@ func NewCmdStudentRepo(f *cmdutil.Factory) *cobra.Command {
 
 			for _, acceptAssignment := range acceptedAssignmentList.AcceptedAssignments {
 				clonePath := filepath.Join(fullPath, acceptAssignment.Repository.Name())
-				fmt.Printf("Cloning into: %v\n", clonePath)
-				_, _, err = gh.Exec("repo", "clone", acceptAssignment.Repository.FullName, "--", clonePath)
-				if err != nil {
-					log.Fatal(err)
-					return
+				if _, err := os.Stat(clonePath); os.IsNotExist(err) {
```
Comment (zrdaley) [inline cmd/gh-classroom/clone/student-repos/student-repos.go:93]: Thank you for adding this check ✨

**Comment by shanep** [comment]
Updated to align with the clone command for both student repos and starter code :)

**Review by shanep — COMMENTED** [review]
I think this should be good to go now. However, please let me know if you still see any issues :)

File: cmd/gh-classroom/pull/student-repos/student-repos.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+				log.Fatal(err)
+			}
+
+			if strings.HasPrefix(directory, "~") {
+				dirname, _ := os.UserHomeDir()
+				directory = filepath.Join(dirname, directory[1:])
+			}
+
+			fullPath, err := filepath.Abs(filepath.Join(directory, assignment.Slug+"-submissions"))
+
```
Comment (zrdaley) [inline cmd/gh-classroom/pull/student-repos/student-repos.go:60]:
```suggestion
```

**Review by zrdaley — APPROVED** [review]
Small nit, but otherwise LGTM! Thank you for this contribution.

**Comment by shanep** [comment]
> Small nit, but otherwise LGTM! Thank you for this contribution.

No worries :) Truly appreciate the feedback! Excited to see this get merged.

**Comment by zrdaley** [comment]
Once https://github.com/github/gh-classroom/actions/runs/6395315615 is green you should be able to upgrade the CLI and use this command.

**Pull request merged**